framework/src/media/MediaPlayerImpl.cpp

```diff
@@ -261,15 +265,7 @@ void MediaPlayerImpl::startPlayer()
 		return;
 	}
 
-	auto prevPlayer = mpw.getPlayer();
-	auto curPlayer = shared_from_this();
-	if (prevPlayer != curPlayer) {
-		if (prevPlayer) {
-			/** TODO Should be considered Audiofocus later **/
-			prevPlayer->pausePlayer();
-		}
-		mpw.setPlayer(curPlayer);
-	}
+	mpw.addPlayer(shared_from_this());
```

Comment:
Assume that there's 2 player, A / B
1. A start PLAY, now list contains 'A'
2. B start Play during A Playing, then A will be paused B will be Added, now list contains 'B' 'A'
3. A start Play, during B playing, then B will paused and A will added again now list contains 'A' 'B' 'A'
. I think making list is good idea but will be added some place before start, also removed when it stopped or terminated.
. In this case there should be logic to find same player in list, but cpu consumption should be considered when it started again from pause state.

Reply:
We have two options;
If user starts A when A is PLAYER_STATE_PLAYING,
1. give error
2. remove A from list and re-add A

In this PR, I selected 1.
We can also select 2, but as you said it takes more cpu consumption(but I think it will be just O(n))